### ISSUE TITLE
Settings dependency uses default values rather than environment variables when used directly

### ISSUE THREAD

**Issue by amagee** [tiangolo/fastapi#2765]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

The [FastAPI docs](https://fastapi.tiangolo.com/advanced/settings/?h=settings#the-main-app-file), when using settings as a dependency, don't depend directly on the `Settings` class but instead on a function that creates an instance of the `Settings` class.

I thought I would try depending on the class directly so as to make things slightly neater and not have to import both the class and the function.

MCVE:

```
from fastapi import FastAPI, Depends
from pydantic import BaseSettings


class Settings(BaseSettings):
    my_setting: str = "default"

def settings():
    return Settings()

app = FastAPI()

@app.get("/small")
async def small(settings: Settings = Depends(settings)):
    """
    Following the pattern from the docs, works as expected.
    """
    print(settings.my_setting)
    return {}

@app.get("/capital")
async def capital(settings: Settings = Depends(Settings)):
    """
    Depend on the class - does not use environment variables.
    """
    print(settings.my_setting)
    return {}
```

Now if I run the server with say, 

```
MY_SETTING=non_default uvicorn app:app --reload
```

then loading the `/small` view will print `non_default`, but loading the `/capital` view will print `default`.

It's unclear to me whether this is intentional; I haven't found anything in the docs that suggests it would behave like this. I would have expected both versions to print "non_default".

### Environment

* OS: Linux
* FastAPI Version: 0.63.0
* Python: 3.8.1

### Additional context

I played around a fair bit trying to track it down. I've got as far as finding that in `solve_dependencies` in `dependencies/utils.py`:

```
    545                 call=call, stack=stack, sub_values=sub_values
    546             )
    547         elif is_coroutine_callable(call):
    548             solved = await call(**sub_values)
    549         else:
--> 550             solved = await run_in_threadpool(call, **sub_values)
    551         if sub_dependant.name is not None:
    552             values[sub_dependant.name] = solved
    553         if sub_dependant.cache_key not in dependency_cache:
    554             dependency_cache[sub_dependant.cache_key] = solved
```

At this point, `sub_values['my_setting']` is `"default"`, which causes the `Settings` object to be constructed with that value.

**Comment by Mause** [comment]
FastAPI doesn't have any integration with the BaseSettings class - it's treating it as it would any other function/callable with default arguments with no annotation. Which is as a dependency that takes query parameters

**Comment by amagee** [comment]
Yeah I didn't expect that there would be any special integration with BaseSettings. But it's still not clear to me why my Settings class when constructed by FastAPI would be constructed differently than if I called `Settings()` myself.

Ie. FastAPI is calling `Settings(my_setting="default")` whereas I would have just expected it to call `Settings()` which would then allow the `BaseSettings` constructor to set the attributes based on the environment variables.

**Comment by Mause** [comment]
As I said in my my comment, FastAPI is treating it as `Settings(my_setting: str = Query("default"))`, IE, like any other dependency

**Comment by tiangolo** [comment]
Thanks for the help here @Mause ! 👏  🙇 

@amagee it seems you are passing `Settings` directly to `Depends()`, you are not supposed to do that.

Please check the docs here: https://fastapi.tiangolo.com/advanced/settings/#settings-in-a-dependency

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
